File: single-language-analysis/dataset-files/python/test_smell_051.py
import pytest
import uuid

class TestHealthITest:
    @pytest.mark.skip
    def test_smell_051(self):
        service_name = str(uuid.uuid4())
        service_id = str(uuid.uuid4())

        client.agent_client().register(8080, 20, service_name, service_id, NO_TAGS, NO_META)
        client.agent_client().pass_health_check(service_id)

        client2 = builder().with_host_and_port(HostAndPort.from_parts("localhost", consul_container.get_first_mapped_port())).build()
        service_id2 = str(uuid.uuid4())

        client2.agent_client().register(8080, 20, service_name, service_id2, NO_TAGS, NO_META)
        client2.agent_client().fail(service_id2)

        response = client2.health_client().get_healthy_service_instances(service_name)
        self.assert_health(service_id, response)

        client.agent_client().deregister(service_id)
        client.agent_client().deregister(service_id2)

    def assert_health(self, service_id, response):
        pass

File: single-language-analysis/dataset-files/java/TestSmell051.java
// Original URL: https://github.com/Vertexvis/consul-client/blob/4c755609e24abab00067d502616c61e61cc3e6f4/src/itest/java/com/orbitz/consul/HealthITest.java#L31-L51

public class TestSmell051 extends BaseIntegrationTest {
    @Test
    @Ignore
    public void shouldFetchPassingNode() throws NotRegisteredException {
        String serviceName = UUID.randomUUID().toString();
        String serviceId = UUID.randomUUID().toString();

        client.agentClient().register(8080, 20L, serviceName, serviceId, NO_TAGS, NO_META);
        client.agentClient().pass(serviceId);

        Consul client2 = builder().withHostAndPort(HostAndPort.fromParts("localhost", consulContainer.getFirstMappedPort())).build();
        String serviceId2 = UUID.randomUUID().toString();

        client2.agentClient().register(8080, 20L, serviceName, serviceId2, NO_TAGS, NO_META);
        client2.agentClient().fail(serviceId2);

        ConsulResponse<List<ServiceHealth>> response = client2.healthClient().getHealthyServiceInstances(serviceName);
        assertHealth(serviceId, response);

        client.agentClient().deregister(serviceId);
        client.agentClient().deregister(serviceId2);
    }
}
